src/stirling/source_connectors/perf_profiler/notes.md

```diff
@@ -0,0 +1,209 @@
+                     +---------------+    |
+                     |  ............ |    |
+                     +---------------+    |
+                     |  return addr. |    |
+-------------------- +---------------+    |
+Leaf fn.    rbp -->  |  saved bp_2   | ---+
+                     +---------------+
+                     |  ............ |
+                     +---------------+
+            rsp -->  |  ............ |
+                     +---------------+
+
+```
+__Figure 1.__ A normal stack (i.e. during function execution, i.e. not at the moment of function entry or function exit).
+
+##### The stack at the moment of function entry or function exit
+```
+-------------------- +---------------+
+Caller_0             |  saved bp_0   |
+                     +---------------+
+                     |  ............ |
+                     +---------------+
+                     |  return addr. |
+-------------------  +---------------+
+Caller_1    rbp -->  |  saved bp_1   |   # points to saved bp_0
+                     +---------------+
+                     |  ............ |
+                     +---------------+
+                     |  return addr. |   # This return address will not get picked up into the stack trace.
+-------------------- +---------------+
+Leaf fn.    rsp -->  |  saved bp_2   |   # points to saved bp_1
+                     +---------------+
+```
+__Figure 2.__ The stack and register state at the moment of function entry or function exit. The stack will be in this state when the instruction pointer points either `push %rbp` (on function entry) or at `ret` (on function exit).
+
+##### Understanding the missing stack frame
+We can see that if register `rpb` is dereferenced with the stack and register state as shown in Figure (2), then `saved bp_2` is skipped and so the return address that points into caller 1 is skipped. Succinctly, the stack is walked back starting from `rbp` which points to `saved bp_1`.
+
+##### Reproducing the "missing stack frames" issue with perf
+
+We collected a few profiles using the Linux perf tool. We reproduced the issue using `--call-graph fp` and confirmed that the issue does not occur when using `--callgraph dwarf` (because dwarf stack trace walking does not use the saved frame pointers, rather, it uses dwarf debug information stored inside of the executable binary).
+```
+# Steps used for perf.
+perf record -F 201 -p <pid> --call-graph <fp|dwarf> sleep 30
+perf script < perf.data > perf.txt
+cat perf.txt | c++filt | gprof2dot.py -f perf -e 0 | dot -Tpdf -o output.pdf
+```
+
+For these experiments, we linked with `ld` vs. our toolchain default of `lld` because [`perf` + `dwarf` does not work with `lld`](https://github.com/llvm/llvm-project/issues/53156).
+
+##### Linux perf users group
+We also sent email to the Linux perf users group and in two different replies, found confirmation of this issue. Editing for brevity:
```

Comment:
Can we add a link to the linux-perf-users group [thread](https://lore.kernel.org/linux-perf-users/CAK2+a3RUFEb4ooe9BehBnBkOKgVt8n4Z28Ryzr2E4SH0--iiWg@mail.gmail.com/T/#t)?